src/details/ArborX_DetailsDistributedSearchTreeImpl.hpp

```diff
@@ -452,21 +452,24 @@ void DistributedSearchTreeImpl<DeviceType>::queryDispatch(
   ////////////////////////////////////////////////////////////////////////////
 }
 
-// FIXME: for some reason Kokkos::BinSort::sort() was not const.
-// If https://github.com/kokkos/kokkos/pull/1310 makes it into master in
-// Trilinos, we might want to pass bin_sort by const reference.
-template <typename BinSort>
-void applyPermutations(BinSort &)
+template <typename PermutationView>
+void applyPermutations(PermutationView const &)
 {
   // do nothing
 }
 
-template <typename BinSort, typename View, typename... OtherViews>
-void applyPermutations(BinSort &bin_sort, View view, OtherViews... other_views)
+template <typename PermutationView, typename View, typename... OtherViews>
+void applyPermutations(PermutationView const &permutation, View view,
+                       OtherViews... other_views)
```

Comment:
I would recommend that these generic functions (applyPermutations and such) to be moved into a utility header (for example, into the one containing sortObjects).

Reply:
Sounds fair, but should still be done, even if in a follow-up PR.

Reply:
Sounds fair, but should still be done, even if in a follow-up PR.